docs/versioning.rst

```diff
@@ -0,0 +1,96 @@
+**********
+Versioning
+**********
+
+
+Current Status
+==============
+
+When a new major version is released it becomes the current release line, where all new features, bug fixes and security patches land.
+
+
+.. list-table:: Release Lines
+   :widths: 20 30 30 20
+   :header-rows: 1
+
+   * - Line
+     - Supported Runtimes
+     - Minimum Trace Agent
+     - Status
+   * - 0.x
+     - CPython: 2.7, 3.5+
+     - 
+     - MAINTENANCE [2022-02-25]
+   * - 1.x
+     - CPython: 2.7, 3.5+
+     - 7.22
+     - CURRENT
+   * - 2.x
+     - TBD
+     - TBD
+     - DEVELOPMENT
+
+
+The **status** of each release line determines when changes are made:
+
+MAINTENANCE
+    No longer under development.
+
+    Receives all bug fixes and security fixes.
+
+    Will not receive any new features.
+
+CURRENT
+    Under active development.
+
+    Receives all backwards compatible new features, bug fixes, and security fixes.
+
+DEVELOPMENT
+    Under active development.
+
+    Receives all incompatible new features.
+
+    Not yet released or maintained.
+
+FUTURE
+    Planned but not under active development.
+
+Interface
+=========
+
+The definition of the **internal** interface:
+    Any module, function, class or attribute is internal if it is prefixed with a single underscore
+
+    Any package that is named ``internal`` is internal
+
+    Any package, module, function, class or attribute that is contained within a namespace that is internal is also internal
+
+The definition of the **public** interface:
+    Any package, module, function, class or attribute not in the internal interface
+
+    Any package, module, function, class or attribute not contained within the ``ddtrace.vendor`` package.
```

Comment:
do we just move this up to the internal definition?

then "public interface" just because "everything else"

Reply:
That makes sense. Any vendor package is also internal.